plugins/modules/nd_site.py

```diff
@@ -0,0 +1,288 @@
+        nd.stdout += str(site_info)
+        if site_info:
+            path = "{0}/{1}".format(path, site)
+            nd.existing = nd.query_obj(path)
+            nd.stdout += str("    PATH : ")
+            nd.stdout += str(path)
+    else:
+        nd.stdout += str("    QUERY ALL : ")
+        nd.existing = nd.query_obj(path).get('items')
+
+
+    nd.previous = nd.existing
+
+    if state == "query":
+        nd.exit_json()
+    elif state == "absent":
+        if nd.existing:
+            if not module.check_mode:
+                nd.stdout += str(" ABSENT  ")
+                nd.stdout += str(path)
+                rm_resp = nd.request(path, method="DELETE")
+                nd.stdout += str(" rm_resp:  ")
+                nd.stdout += str(rm_resp)
+                if rm_resp is {}:
+                    nd.existing = {}
+            nd.existing = {}
+    elif state == "present":
+        payload ={
+            "spec": {
+                "name": site,
+                "siteType": SITE_TYPE_MAP.get(site_type),
+                "host": url,
+                "userName": site_username,
+                "password": site_password,
+                "loginDomain": login_domain,
+                "loginDomain": "DefaultAuth",
+                "latitude": str(latitude),
+                "longitude": str(longitude),
+                # "securityDomains": security_domains if security_domains is not None else [],
+                "siteConfig": {
+                    "aci": {
+                        "InbandEPGDN": inband_epg,
+                    }
+            }
+            },
+        }
+        nd.stdout += str("    PAYLOAD : ")
+        nd.stdout += str(payload)
+
+        nd.sanitize(payload, collate=True)  # check it is applies everything or just modified value.
+
+        if not module.check_mode:
+            method = "POST"
+            if nd.existing:
+                method = "PUT"
+                nd.stdout += str("    IF_ND.EXISTING path : ")
+                nd.stdout += str(path)
+
+            nd.stdout += str("    METHOD : ")
+            nd.stdout += str(method)
+            # unwanted = ["url", ["aci", "userName"], ["aci", "password"], ["aci", "loginDomain"], ["aci", "switches"], ["aci", "appUserName"]]
+            unwanted = [
+                "metadata",
+                "status",
+                ["spec", "annotation"],
+                ["spec", "host"],
+                ["spec", "userName"],
+                ["spec", "password"],
+                ["spec", "loginDomain"],
+                ["spec", "useProxy"],
+                ["spec", "secureVerify"],
+                ["spec", "internalOnboard"]
+            ]
+            if nd.get_diff(unwanted):
+                nd.existing = nd.request(path, method=method, data=payload)
+            else:
+                nd.has_modified = True
```

Comment:
What's the difference between the two - nd.has_modified = True and nd.result["changed"] = False?
Why are we setting both?